inference-engine/src/transformations/src/transformations/smart_reshape/mimic_set_batch_size.cpp

```diff
@@ -0,0 +1,113 @@
+        const auto & old_output_batch = static_cast<float>(pattern_vector[0]);
+
+        const auto & scale = old_output_batch / old_input_batch;
+
+        const auto & shape_of = std::make_shared<opset5::ShapeOf>(reshape->get_input_source_output(0), pattern->get_element_type());
+        const auto & new_input_batch = std::make_shared<ngraph::opset5::Gather>(
+                shape_of, ngraph::opset5::Constant::create(ngraph::element::i64, {1}, std::vector<int64_t>{0}),
+                ngraph::opset5::Constant::create(ngraph::element::i64, {}, std::vector<int64_t>{0}));
+
+        const std::shared_ptr<Node> & new_output_batch = std::make_shared<opset5::Convert>(
+                std::make_shared<opset5::Ceiling>(
+                        std::make_shared<opset5::Multiply>(
+                                std::make_shared<opset5::Convert>(new_input_batch, element::f32),
+                                opset5::Constant::create(element::f32, {1}, {scale}))),
+                pattern->get_element_type());
+
+        auto new_reshape_pattern = new_output_batch;
+        const auto rank = pattern_vector.size();
+        if (rank > 1) {
+            std::vector<int64_t> non_batch_dims(rank - 1);
+            std::iota(non_batch_dims.begin(), non_batch_dims.end(), 1);
+            const auto & non_batch_dims_node = std::make_shared<ngraph::opset5::Gather>(
+                    pattern,
+                    ngraph::opset5::Constant::create(ngraph::element::i64, {non_batch_dims.size()}, non_batch_dims),
+                    ngraph::opset5::Constant::create(ngraph::element::i64, {}, std::vector<int64_t>{0}));
+            new_reshape_pattern = std::make_shared<opset5::Concat>(OutputVector{new_reshape_pattern, non_batch_dims_node}, 0);
+        }
+        reshape->input(1).replace_source_output(new_reshape_pattern->output(0));
+        return true;
+    };
+    auto m = std::make_shared<ngraph::pattern::Matcher>(reshape_label, "MimicSetBatchSize");
+    register_matcher(m, callback);
+}
+
+
+void set_folding_for_PriorBox(std::shared_ptr<ngraph::Node> prior_box, bool flag) {
+    std::string rt_info_disable_cf = "DISABLED_CONSTANT_FOLDING";
+    std::vector<ngraph::Node::type_info_t> allowed_to_skip = {
+            ngraph::opset1::Convert::type_info,
+            ngraph::opset1::StridedSlice::type_info,
+    };
+    std::vector<ngraph::Node::type_info_t> types_to_find = {
+            ngraph::opset1::ShapeOf::type_info,
+            ngraph::opset3::ShapeOf::type_info,
+    };
+
+    std::deque<std::shared_ptr<ngraph::Node>> nodes;
+    nodes.push_back(prior_box->get_input_node_shared_ptr(0));
+    nodes.push_back(prior_box->get_input_node_shared_ptr(1));
+
+    while (!nodes.empty()) {
+        auto curr_node = nodes.front();
+        nodes.pop_front();
+        if (count(allowed_to_skip.begin(), allowed_to_skip.end(), curr_node->get_type_info())) {
```

Comment:
With `unordered_set`
```suggestion
        if (allowed_to_skip.count(curr_node->get_type_info())) {
```